src/main/java/com/google/graphgeckos/dashboard/storage/DatastoreRepository.java

```diff
@@ -160,4 +162,40 @@ public BuildInfo getRevisionEntry(String commitHash) throws IllegalArgumentExcep
 
     return storage.findById(commitHash, BuildInfo.class);
   }
+
+  /**
+   * {@inheritDoc}
+   */
+  int getBuildbotIndex(String buildbotName) throws IllegalArgumentException, NotBoundException {
+    if (buildbotName == null) {
+      throw new IllegalArgumentException("buildbotName cannot be null");
+    }
+
+    BuilderIndex associatedEntity = storage.findById(buildbotName, BuilderIndex.class);
+
+    if (associatedEntity == null) {
+      throw new NotBoundException("buildbotName not bound to any index");
+    }
+
+    return associatedEntity.getIndex();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  void setBuildbotIndex(String buildbotName, int newValue) throws IllegalArgumentException,
```

Comment:
You are testing only with positive values. Are negative indices accepted as well? If yes, please add a test showing that it's valid. The documentation in pull request #121 doesn't speak to this either.

Reply:
While negative indices should be caught under the "not <= than the previous value", I added this condition to throw Index OOB in #121's documentation, since this could happen without triggering the exception for the first creation of an entry.